app/assets/javascripts/discourse/app/components/form-template-field/wrapper.hbs

```diff
@@ -1,5 +1,8 @@
 {{#if this.parsedTemplate}}
-  <div class="form-template-form__wrapper">
+  <div
+    class="form-template-form__wrapper"
+    {{did-update this.refreshTemplate @id}}
+  >
```

Comment:
`@id` must be passed here.

When passed it returns the `<div class="form-template-form__wrapper">` element inside `this.refreshTemplate(id)` when logging out the `id`, so we can't really use that value. However, it needs to still be passed because `did-update` will only account for attribute changes if a parameter is passed.

See: https://github.com/emberjs/ember-render-modifiers#example-resizing-text-area
>One key thing to know about {{did-update}} is it will not rerun whenever the contents or attributes on the element change. For instance, {{did-update}} will not rerun when @type changes here:

> If {{did-update}} should rerun whenever a value changes, the value should be passed as a parameter to the modifier. For instance, a textarea which wants to resize itself to fit text whenever the text is modified could be setup like this:

Reply:
Yes I know very well how it works :) you were only speaking of {{did-update}} since the beginning that's why I asked what you exactly tried